encoding.go

```diff
@@ -81,6 +86,10 @@ func (k *SpanIdentity) Dims() map[string]string {
 	if k.ServiceMesh {
 		m["sf_serviceMesh"] = "true"
 	}
+	if k.AdditionalDimensions != "" {
+		m["sf_dimensionalized"] = "true"
+		json.Unmarshal([]byte(k.AdditionalDimensions), &m)
```

Comment:
I added a constructor feel free to do something else, but my main concern here is the ability to override any of the fields we already set (include sf_dimenzionalized). We need to forbid this either in the constructor or the dims method. Probably the constructor since if we already have a histo for these and we're not emitting metrics, it seems like a bad idea. I added a test that fails because of this.

Reply:
I tweaked the Dims() method to ensure we unmarshal `AdditionalDims` first to the map, and then set all `SpanIdentity` properties on top of the `AdditionalDims`... this prevents additional dims from overwriting.  It doesn't prevent a situation where `httpMethod` is set in `AdditionalDims`, but not set in `SpanIdentity.HttpMethod`.  Should we safeguard for that? It seems rare to me.